### ISSUE TITLE
OpenMPI cannot handle standard UNIX filename

### ISSUE THREAD

**Issue by nasailja** [open-mpi/ompi#860]
This program returns 3 instead of 0 because openmpi freaks out about a : in a filename:
```
#include "stdio.h"
#include "mpi.h"
int main(int argc, char* argv[]){
	/*FILE* f1 = fopen(":", "w");
	if (f1 == NULL) { return 1; };*/

	if (MPI_Init(&argc, &argv) != MPI_SUCCESS) { return 2; }
	MPI_File outfile;
	if (MPI_File_open(
		MPI_COMM_WORLD,
		":",
		MPI_MODE_CREATE | MPI_MODE_WRONLY,
		MPI_INFO_NULL,
		&outfile
	) != MPI_SUCCESS) { return 3; }
	return 0;
}
```
The error I get from another program, where : is part of a directory name, is:`MPI_ERR_NO_SUCH_FILE: no such file or directory`

According to MPI-2.0 the file name format must be documented and the manpage at https://www.open-mpi.org/doc/v1.8/man3/MPI_File_open.3.php seems like a good place to mention this openmpi feature.
I'm using mpiexec (OpenRTE) 1.8.8 on OSX.

**Comment by edgargabriel** [comment]
It is unfortunately a bit more complicated. The ROMIO implementation of MPI I/O (which is the default in the 1.8 series) interpretes to the best of my knowledge the ":" character, since it can be used to provide additional information, e.g. file system type, see page 494 of the MPI 3.1 document 

An implementation may require that filename include a
string or strings specifying additional information about the file. Examples include
the type of filesystem (e.g., a prefix of ufs:), a remote hostname (e.g., a prefix of
machine.univ.edu:), or a file password (e.g., a suffix of /PASSWORD=SECRET). (End
of advice to implementors.)

OMPIO, does not interpret the symbol, and creating a file called ":" actually works, I just tested it.  So the documentation would have to go into some details on that.

**Comment by nasailja** [comment]
Thanks, at least docs could warn that some configurations also do not allow characters other than the usual / and NULL. configure states:
```
  --disable-mpi-io        Disable built-in support for MPI-2 I/O, likely
                          because an externally-provided MPI I/O package will
                          be used. Default is to use the internal component
                          system and its specially modified version of ROMIO
  --disable-io-romio      Disable the ROMIO MPI-IO component
```
so what is left if I disable romio, MPI-1 I/O only?

**Comment by edgargabriel** [comment]
no, you don't have to disable it, you could try to switch to ompio, e.g.

mpiexec --mca io ompio -np 4 ./yourtest

Note, that the ompio version in 1.8.8 has some restrictions (the version in master and in the upcoming 2.x series is much more stable), but you could give it a shot.

**Comment by nasailja** [comment]
Thanks that worked and didn't seem to affect performance.

**Comment by jsquyres** [comment]
@edgargabriel Does OMPIO reserve any characters in the filename?  We should probably update the man page for MPI_File_open(3) for both OMPI and ROMIO, as @nasailja suggests.

**Comment by nasailja** [comment]
You can use https://github.com/open-mpi/ompi/pull/861 ...

**Comment by edgargabriel** [comment]
Not that I am aware of, but honestly before running the test of @nasailja I was not entirely sure about it :-)

**Comment by jsquyres** [comment]
Sorry for the delay -- I finally made a comment on the suggested new message in #861.
